lib/chai/core/assertions.js

```diff
@@ -536,6 +536,13 @@ module.exports = function (chai, _) {
       case 'string':
         itemsCount = val.length;
         break;
+      case 'map':
+      case 'set':
+        itemsCount = val.size;
+        break;
+      case 'weakmap':
+      case 'weakset':
+        throw new TypeError('.empty was called on a weak collection');
```

Comment:
Right, thanks, fixed that. Another error message was just 3 lines away, what's wrong with me...